org/pr/installable-build.ts

```diff
@@ -0,0 +1,142 @@
+    console.log(`No open PR found for this commit ${status.commit.sha}`)
+    return []
+  }
+  return prsWithCommit
+}
+
+export async function createOrUpdateComment(status, message) {
+  const api = danger.github.api
+  const owner = status.repository.owner.login
+  const repo = status.repository.name
+
+  // Add a prefix to our comment so we can easily find/update it later.
+  const commentPrefix = "<!--- Installable Build Comment --->"
+  const commentBody = `${commentPrefix}\n${message}`
+
+  // Since a commit can be on more than one PR, find all PRs that have this commit
+  const prsWithStatus = await getPRsWithStatus(status)
+  for (const number of prsWithStatus) {
+    const pull = await api.pulls.get({ owner, repo, number })
+    if (pull.data.head.sha !== status.commit.sha) {
+      console.log(`${status.commit.sha} is not the latest commit on PR ${number}, skipping comment`)
+      continue
+    }
+
+    const allComments = await api.issues.listComments({owner, repo, number})
+
+    const existingComment = allComments.data.find(comment => comment.user.id === PERIL_BOT_USER_ID && comment.body.includes(commentPrefix))
+    let commentResult
+    if (existingComment !== undefined) {
+      commentResult = await api.issues.updateComment({
+        owner, 
+        repo, 
+        comment_id: existingComment.id,
+        body: commentBody
+      })
+    } else {
+      commentResult = await api.issues.createComment({
+        owner,
+        repo,
+        number,
+        body: commentBody
+      })
+    }
+
+    console.log(`Installable build comment posted to ${commentResult.data.html_url}`)
+  }
+}
+
+async function circleCIArtifacts(status) {
+  // CircleCI URLs look like https://circleci.com/gh/:org/:repo/12345?some=query'
+  // We need to extract the build number
+  const urlPath = url.parse(status.target_url).pathname // Gives the /gh/:org/:repo/12345 portion
+  const buildNumber = +path.parse(urlPath).base
+
+  const owner = status.repository.owner.login
+  const repo = status.repository.name
+
+  const artifactsUrl = `https://circleci.com/api/v1.1/project/gh/${owner}/${repo}/${buildNumber}/artifacts?circle-token=${CIRCLECI_TOKEN}`
+
+  const res = await fetch(artifactsUrl)
+  if (res.ok) {
+    return res.json()
+  }
+  return []
+}
+
+async function getDownloadCommentText(status) {
+  const artifacts = await circleCIArtifacts(status)
+
+  const commentJsonArtifact = artifacts.find(artifact => artifact.path.endsWith("comment.json"))
+  if (commentJsonArtifact) {
+    // Download the JSON file so we can get the comment text
+    const res = await fetch(commentJsonArtifact.url)
+    if (res.ok) {
+      return res.json()["body"]
+    }
+  }
+
+  const apkArtifact = artifacts.find(artifact => artifact.path.endsWith(".apk"))
+  if (apkArtifact) {
+    return `You can test the changes on this Pull Request by downloading the APK [here](${apkArtifact.url}).`
+  }
+  return undefined
+}
+
+export default async (status: Status) => {
```

Comment:
Peril runs the default export. I have extracted most of the logic into functions to hopefully make it easy to follow.